app/models/role.rb

```diff
@@ -17,6 +18,14 @@ def self.add_admin(person)
     find_or_create_by(person_id: person.id, name: "admin")
   end
 
+  def self.moderator?(person)
+    exists?(person_id: person.id, name: "moderator")
```

Comment:
This should be `moderators.exists?(person_id: person.id)` and then the `|| current_user.admin?` above shouldn't be needed anymore.